config/config.go

```diff
@@ -1545,94 +1547,125 @@ func migrateConfig(configFile, targetDir string) (string, error) {
 	return target, nil
 }
 
-// ReadConfig verifies and checks for encryption and verifies the unencrypted
-// file contains JSON.
-// Prompts for decryption key, if target file is encrypted
+// ReadConfig reads the configuration from the given file
+// if target file is encrypted, prompts for encryption key
+// Also - if not in dryrun mode - it checks if the configuration needs to be encrypted
+// and stores the file as encrypted, if necessary (prompting for enryption key)
 func (c *Config) ReadConfig(configPath string, dryrun bool) error {
 	defaultPath, _, err := GetFilePath(configPath)
 	if err != nil {
 		return err
 	}
-
-	fileData, err := ioutil.ReadFile(defaultPath)
+	confFile, err := os.Open(defaultPath)
 	if err != nil {
 		return err
 	}
+	defer confFile.Close()
+	result, wasEncrypted, err := ReadConfig(confFile, func() ([]byte, error) { return PromptForConfigKey(false) })
```

Comment:
I think it would be better to rename one of these functions. Having two `ReadConfig`'s in the one package that do different things will make it confusing, even if one is belonging to `Config`

Reply:
The new ReadConfig is a thread-safe constructor (alternative to the weird singleton approach `c:=&config.Cfg; c.ReadConfig(...)`) for creating config objects. What do you propose? ReadJson? NewFromJson?

Reply:
With the context of your other comments, I think perhaps `ReadConfigFromFile`?

Reply:
renamed the old ReadConfig